File: LeetCode_Marathons/LCM4/Easy/Find_the_Pivot_Integer/a.go
func pivotInteger(n int) int {
    sumN := n * (n + 1) / 2
    dpSum := 0
    for i := 1; i <= n; i++ {
        dpSum += i
        if dpSum == sumN - dpSum + i {
            return i
        }
    }
    return -1
}

File: LeetCode_Marathons/LCM4/Easy/Find_the_Pivot_Integer/a.py
class Solution:
    def pivotInteger(self, n: int) -> int:
        sumN: int = n * (n + 1) // 2
        dpSum: int = 0
        for i in range(1, n + 1):
            dpSum += i
            if dpSum == sumN - dpSum + i: return i
        return -1